api/v1/openapi.json

```diff
@@ -0,0 +1,1115 @@
+                                                }
+                                        },
+                                        "default": {
+                                                "description": "Unexpected error",
+                                                "schema": {
+                                                        "$ref": "#/definitions/Error"
+                                                }
+                                        }
+				}
+			},
+			"delete": {
+				"description": "Delete an endpoint",
+				"tags": [ "endpoint" ],
+				"parameters": [
+					{ "$ref": "#/parameters/endpoint-id" },
+					{ "$ref": "#/parameters/endpoint-id-type" }
+				],
+				"responses": {
+					"200": {
+                                                "description": "Success",
+                                                "schema": {
+                                                        "$ref": "#/definitions/Endpoint"
+                                                }
+					},
+                                        "404": {
+                                                "description": "Endpoint not found",
+                                                "schema": {
+                                                        "$ref": "#/definitions/Error"
+                                                }
+                                        }
+				}
+			}
+		},
+                "/endpoint/list": {
+			"get": {
+				"summary": "Retrieve list of all endpoints",
+				"tags": [ "endpoint" ],
+				"responses": {
+					"200": {
+                                                "description": "Success",
+                                                "schema": {
+							"type": "array",
+						        "items": {
+								"$ref": "#/definitions/Endpoint"
+							}
+                                                }
+					}
+				}
+			}
+		},
+                "/endpoint/config": {
+			"get": {
+				"summary": "Retrieve endpoint configuration",
+				"tags": [ "endpoint" ],
+				"parameters": [
+					{
+						"name": "id",
+						"description": "Endpoint ID",
+						"in": "query",
+						"required": true,
+						"type": "integer"
+					}
+				],
+				"responses": {
+					"200": {
+                                                "description": "Success",
+                                                "schema": {
+                                                        "$ref": "#/definitions/ConfigurationResponse"
+                                                }
+					},
+                                        "404": {
+                                                "description": "Endpoint not found",
+                                                "schema": {
+                                                        "$ref": "#/definitions/Error"
+                                                }
+                                        }
```

Comment:
It is specifically left out if no unexpected error may occur, e.g. `GET `/endpoint/labels/` is either success or 404. There are no other errors. Will double check this everywhere though.